azurelinuxagent/common/osutil/default.py

```diff
@@ -901,7 +901,10 @@ def get_endpoint_from_leases_path(pathglob):
         return endpoint
 
     def is_missing_default_route(self):
-        routes = shellutil.run_get_output("route -n")[1]
+        route_cmd = "route -n"
+        if not shellutil.has_command("route"):
```

Comment:
It is inevitable that other distributions also drop net-tools. Creating distribution specific implementations as suggested implies that we'd end up with one obsolete default implementation and number_of_distros.py files, from my perspective that is not great especially since the implementation in number_of_distros.py is the same.

There are other means of doing this kind of indirection, but creating a file for each distro is IMHO not the way to go.

Reply:
But that's the way the agent is architected. Distro-specific differences are abstracted. When the majority moves away from net-tools, the default implementation would be changed to reflect that.

Reply:
I would like to centralize this behavior too. How about a `networkutil.py` in the same vein as `textutil.py`. There are three commands to add: add, del, and show.  The actions and probing for the command can happen in this class. In the future, we can revisit this case and have to refactor one location, and any platform specific code will be contained.

I would add networkutil.py under azurelinuxagent/common/utils.

Reply:
I can live with this, especially if you probe only once (when the first add/del/show is done) and rely on that decision for the remaining life of the process. A "factory" method that computes, caches, and returns the correct class instance (net-utils or "ip route") seems the most logical approach.